cg/cli/demultiplex/copy_novaseqx_data.py

```diff
@@ -0,0 +1,82 @@
+from pathlib import Path
+import os
+import shutil
+from typing import List, Optional
+
+from cg.constants.demultiplexing import DemultiplexingDirsAndFiles
+
+
+def is_demultiplexing_copied(analysis_directory: Path) -> bool:
+    """Determine whether the demultiplexing has been copied for the latest analysis for a novaseqx flow cell."""
+    return Path(analysis_directory, DemultiplexingDirsAndFiles.COPY_COMPLETE).exists()
+
+
+def is_flow_cell_demultiplexed(analysis_directory: Path) -> bool:
+    return Path(
+        analysis_directory,
+        DemultiplexingDirsAndFiles.DATA,
+        DemultiplexingDirsAndFiles.ANALYSIS_COMPLETED,
+    ).exists()
+
+
+def is_flow_cell_in_demultiplexed_runs(flow_cell_name: str, demultiplexed_runs: Path) -> bool:
+    return Path(demultiplexed_runs, flow_cell_name).exists()
+
+
+def get_latest_analysis_directory(flow_cell_dir: Path) -> Optional[Path]:
+    analysis_path = Path(flow_cell_dir, DemultiplexingDirsAndFiles.ANALYSIS)
+
+    if not analysis_path.exists():
+        return None
+    analysis_versions = get_sorted_analysis_versions(analysis_path)
+    return analysis_versions[0] if analysis_versions else None
+
+
+def get_sorted_analysis_versions(analysis_path: Path) -> List[Path]:
+    def sort_by_name(version: Path) -> int:
+        return int(version.name)
+
+    analysis_versions = [version for version in analysis_path.iterdir() if version.is_dir()]
+    sorted_versions = sorted(analysis_versions, key=sort_by_name, reverse=True)
+    return sorted_versions
+
+
+def is_queued_for_post_processing(flow_cell_dir: Path) -> bool:
+    return Path(flow_cell_dir, DemultiplexingDirsAndFiles.QUEUED_FOR_POST_PROCESSING).exists()
+
+
+def copy_flow_cell_analysis_data(flow_cell_dir: Path, destination: Path) -> None:
+    analysis: Path = get_latest_analysis_directory(flow_cell_dir)
+    analysis_data = Path(analysis, DemultiplexingDirsAndFiles.DATA)
+
+    hardlink_tree(src=analysis_data, dst=destination)
```

Comment:
```suggestion
    hardlink_tree(src=Path(analysis, DemultiplexingDirsAndFiles.DATA), dst=destination)
```
Do we need the extra variable? I think the Path is very clear

Reply:
Subjective, but I think it is more readable as a separate variable